### PR TITLE
Feature/Product Grid Section - UI

### PR DESCRIPTION
**Pull request by Umasankar1522**

### CODE REVIEW

File: src/backend/productsList.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+export type productsDataType = {
```
Comment (prateekoli9) [inline src/backend/productsList.ts:1]: Move to types/productList.ts file

File: src/backend/productsList.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    imageUrl: string;
+    imageUrlOnHover: string;
+    sizes: number[];
+    productTitle: string;
+    productSubTitle: string;
+    productPrice: number;
+    productId: number;
+};
+
+const productsList: productsDataType[] = [
```
Comment (prateekoli9) [inline src/backend/productsList.ts:11]: Try to make it similar to existing response

File: src/components/ProductsGridComponent/ProductsGrid.component.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    const onImageLeaveHandler = (): void => {
+        setOnImageHovered(false);
+    };
+
+    return (
+        <Grid item key={productId} md={4} lg={3} sm={6} xs={6} sx={{ marginBottom: '15px' }}>
+            <Grid container sx={{ position: 'relative' }}>
+                <a href=''>
+                    {!onImageHovered &&<img style={{ display: 'block', width: '100%', height: '100%' }}
```
Comment (prateekoli9) [inline src/components/ProductsGridComponent/ProductsGrid.component.tsx:20]: Change to src={onImageHovered ? hoveredSrc : imageSrc}

File: src/components/ProductsGridComponent/ProductsGrid.component.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import React, { FC, useState } from 'react';
```
Comment (prateekoli9) [inline src/components/ProductsGridComponent/ProductsGrid.component.tsx:1]: Change component name to ProductListItem

File: src/pages/product-list/product-list.component.tsx [hunk 5]
```diff
@@ -1,29 +1,38 @@
...
-              <div className={styles.productList__Content__Products__Header}>
-                 <p> Showing 6514 Result(s) in 62ms </p>
-                 <p>Sort Component</p>
-              </div>
-              <div className={styles.productList__Content__Products__Header__Data}>
-               
-              </div>
+          <div className={styles.productList__Content__Products__Header__Data}>
+            <Grid container spacing={1}>
+            { products.map((product: productsDataType): JSX.Element => ( <ProductsGrid key={product.productId} {...product}/> )) }
```
Comment (prateekoli9) [inline src/pages/product-list/product-list.component.tsx:30]: Change component name

File: src/pages/product-list/product-list.module.scss [hunk 6]
```diff
@@ -28,10 +28,13 @@
                 padding: 0 15px;
 
                 &__Data{
-                    display: grid;
-                    grid-template-columns: 30% 30% 30%;
-                    justify-content: center;
-                    grid-gap: 1rem;
+                    // display: grid;
```
Comment (prateekoli9) [inline src/pages/product-list/product-list.module.scss:31]: Remove commented code

Comment (Umasankar1522) [reply src/pages/product-list/product-list.module.scss]: Removed the commented code

File: src/components/ProductsGridComponent/ProductsGrid.component.tsx (hunk 4 above)
Comment (Umasankar1522) [reply src/components/ProductsGridComponent/ProductsGrid.component.tsx:1]: Renamed ProductGrid to ProductListItem

File: src/pages/product-list/product-list.component.tsx (hunk 5 above)
Comment (Umasankar1522) [reply src/pages/product-list/product-list.component.tsx]: Renamed ProductGrid to ProductListItem

File: src/components/ProductsGridComponent/ProductsGrid.component.tsx (hunk 3 above)
Comment (Umasankar1522) [reply src/components/ProductsGridComponent/ProductsGrid.component.tsx:20]: As suggested, used ternary operators to display the image based on the onImageHovered state.

File: src/backend/productsList.ts (hunk 2 above)
Comment (Umasankar1522) [reply src/backend/productsList.ts:11]: removed ProductList and created a new Object (ProductListItem) as per the properties in existing response object

File: src/backend/productsList.ts (hunk 1 above)
Comment (Umasankar1522) [reply src/backend/productsList.ts]: moved into component as I am using only in that product-list.component

**Pull request merged**